protocol/contracts/v2/templeLineOfCredit/TempleLineOfCredit.sol

```diff
@@ -0,0 +1,968 @@
+
+                // Clear the account data
+                delete allAccountsData[_account];
+            }
+        }
+
+        // burn the temple collateral by repaying to TRV. This will burn the equivalent dUSD debt too.
+        if (totalCollateralClaimed != 0) {
+            templeToken.safeIncreaseAllowance(address(treasuryReservesVault), totalCollateralClaimed);
+            treasuryReservesVault.repayTemple(totalCollateralClaimed, address(tlcStrategy));
+            totalCollateral -= totalCollateralClaimed;
+        }
+
+        // Remove debt from the totals
+        repayTotalDebt(_debtTokenCache, totalDebtWiped.encodeUInt128());
+    }
+    
+    /*´:°•.°+.*•´.*:˚.°*.˚•´.°:°•.°•.*•´.*:˚.°*.˚•´.°:°•.°+.*•´.*:*/
+    /*                            ADMIN                           */
+    /*.•°:°.´+˚.*°.˚:*.´•*.+°.•°:´*.´•*.•°.•°:°.´:•˚°.*°.˚:*.´+°.•*/
+
+    /**
+     * @notice Update the TLC Strategy contract, and Treasury Reserves Vault (TRV)
+     * @dev The TRV is granted access to spend DAI, in order to repay debt.
+     */
+    function setTlcStrategy(
+        address newTlcStrategy
+    ) external override onlyElevatedAccess {
+        tlcStrategy = ITlcStrategy(newTlcStrategy);
+
+        // Remove allowance from the old TRV
+        address previousTrv = address(treasuryReservesVault);
+        if (previousTrv != address(0)) {
+            daiToken.safeApprove(previousTrv, 0);
+        }
+
+        address _trv = address(tlcStrategy.treasuryReservesVault());
+        treasuryReservesVault = ITreasuryReservesVault(_trv);
+
+        // Set max allowance on the new TRV
+        {
+            daiToken.safeApprove(_trv, 0);
+            daiToken.safeIncreaseAllowance(_trv, type(uint256).max);
+        }
+
+        emit TlcStrategySet(newTlcStrategy, _trv);
+
+        // The new TRV may have a different debt ceiling. Force an update to the interest rates.
+        updateInterestRates(debtTokenCache());
+    }
+
+    /**
+     * @notice Set the Withdrawal Collateral Request window parameters
+     * @param minSecs The number of seconds which must elapse between a request and the action
+     * @param maxSecs The number of seconds until a request expires
+     */
+    function setWithdrawCollateralRequestConfig(
+        uint256 minSecs,
+        uint256 maxSecs
+    ) external override onlyElevatedAccess {
+        emit RemoveCollateralRequestConfigSet(minSecs, maxSecs);
+        removeCollateralRequestConfig = FundsRequestConfig(uint32(minSecs), uint32(maxSecs));
+    }
+
+    /**
+     * @notice Set the Borrow Request window parameters
+     * @param minSecs The number of seconds which must elapse between a request and the action
+     * @param maxSecs The number of seconds until a request expires
+     */
+    function setBorrowRequestConfig(
+        uint256 minSecs,
+        uint256 maxSecs
+    ) external override onlyElevatedAccess {
+        emit BorrowRequestConfigSet(minSecs, maxSecs);
```

Comment:
revert if `minSecs >= maxSecs`

Reply:
Not required anymore since the cooldown was rugged